pkg/controller/sync/controller.go

```diff
@@ -378,7 +378,11 @@ func (s *KubeFedSyncController) syncToClusters(fedResource FederatedResource) ut
 	}
 
 	collectedStatus := dispatcher.CollectedStatus()
-	return s.setFederatedStatus(fedResource, status.AggregateSuccess, &collectedStatus)
+	status := s.setFederatedStatus(fedResource, status.AggregateSuccess, &collectedStatus)
+	if !ok {
+		return util.StatusNeedsRecheck
+	}
```

Comment:
It is preferable to make use of the `ok` variable right after it has been assigned, i.e. on line 370. This way it's much easier to follow the reasoning of why it is assigned in the first place.